### PR TITLE
#362 Add unit tests and React Testing Library customRender function

### LINKED ISSUE #362: Add React Testing Library, Set up a custom render function
**Related User Story**

Add React Testing Library and set up a custom render function as the docs recommend to 
provide global context provides and data stores in addition to any other necessary configurations

**Related Tickets**

**Acceptance Criteria**

One working unit test with the custom render function

**Subtasks**
<!-- If necessary, break down the ticket into smaller, actionable subtasks. Create issues accordingly.>

### PR DESCRIPTION
**Pull request by colinvandergraaf**
- **Add RTL and custom render function**
- **Add RTL and custom render function; add App unit test**
- **Modify customRender function to use memoryRouter; Add tests for 404 page**
- **Update NotFoundPage unit test**
- **Update routeConfig in testUtils; typo**

## Description
This PR adds [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) which helps us render React components and create unit tests that better align with how users use our application.  

RTL recommends setting up a[ customRender function ](https://testing-library.com/docs/react-native-testing-library/setup/#custom-render) to act as a centralized place to include global context providers and data stores.  Our customRender function is in `testUtils.tsx` which  provides our application Providers and routing to React Testing Library in one place.  

Note: When you write unit tests you should import the render function and RTL variables  from `testUtils.tsx` and not the imported package.  

The vitest.setup.ts  extends vitest with jest-dom  matchers to better test React components with React Testing Library and handling test cleanup in a centralized location.  

 

Fixes  #362 #341 

## Type of change

- [x]  🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨  New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Can this be tested? Testing Plan to review this PR
- checkout the branch and run `npm install `
- navigate to the client directory  and run `npm test`
- see tests for App.unit.test.tsx and NotFoundPage.unit.test.tsx pass 

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: client/routes/index.tsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
-import React from 'react';
-import { type RouteObject, createBrowserRouter } from 'react-router-dom';
+// routes/index.ts
```
Comment (kinskiwu) [inline client/routes/index.tsx]: I think you forgot to move line 1

File: client/tests/global.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+/// <reference types="vitest" />
```
Comment (kinskiwu) [inline client/tests/global.d.ts]: Maybe we can move this file to the`/types` folder?

File: client/tests/unit/App.unit.test.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// App.unit.test.tsx
```
Comment (kinskiwu) [inline client/tests/unit/App.unit.test.tsx]: Think you forgot to remove line 1

**Review by kinskiwu — APPROVED** [review]
Thanks for the setup! Ran it and all the tests passed. Just very minor issues. Thank you for the good work as always.

File: client/tests/global.d.ts (hunk 2 above)
Comment (colinvandergraaf) [reply client/tests/global.d.ts]: good idea!

File: client/tests/unit/App.unit.test.tsx (hunk 3 above)
Comment (colinvandergraaf) [reply client/tests/unit/App.unit.test.tsx]: thanks for the code review!

**Review by chrisr0892 — APPROVED** [review]
Looks good!

**Pull request merged**